test/e2e/framework/statefulset/fixtures.go

```diff
@@ -0,0 +1,278 @@
+	}
+
+	vols := []corev1.Volume{}
+	for _, m := range podMounts {
+		vols = append(vols, corev1.Volume{
+			Name: m.Name,
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: fmt.Sprintf("/tmp/%v", m.Name),
+				},
+			},
+		})
+	}
+
+	return &appsv1.StatefulSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "StatefulSet",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Replicas: func(i int32) *int32 { return &i }(replicas),
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      labels,
+					Annotations: map[string]string{},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:         "nginx",
+							Image:        imageutils.GetE2EImage(imageutils.Nginx),
+							VolumeMounts: mounts,
+						},
+					},
+					Volumes: vols,
+				},
+			},
+			UpdateStrategy:       appsv1.StatefulSetUpdateStrategy{Type: appsv1.RollingUpdateStatefulSetStrategyType},
+			VolumeClaimTemplates: claims,
+			ServiceName:          governingSvcName,
+		},
+	}
+}
+
+// NewStatefulSetPVC returns a PersistentVolumeClaim named name, for testing StatefulSets.
+func NewStatefulSetPVC(name string) corev1.PersistentVolumeClaim {
+	return corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.ReadWriteOnce,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: *resource.NewQuantity(1, resource.BinarySI),
+				},
+			},
+		},
+	}
+}
+
+// CreateStatefulSetService creates a Headless Service with Name name and Selector set to match labels.
+func CreateStatefulSetService(name string, labels map[string]string) *corev1.Service {
+	headlessService := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+		},
+	}
+	headlessService.Spec.Ports = []corev1.ServicePort{
+		{Port: 80, Name: "http", Protocol: corev1.ProtocolTCP},
+	}
+	headlessService.Spec.ClusterIP = "None"
+	return headlessService
+}
+
+// SetHTTPProbe sets the pod template's ReadinessProbe for Nginx StatefulSet containers.
+// This probe can then be controlled with BreakHTTPProbe() and RestoreHTTPProbe().
+// Note that this cannot be used together with PauseNewPods().
+func SetHTTPProbe(ss *appsv1.StatefulSet) {
+	ss.Spec.Template.Spec.Containers[0].ReadinessProbe = httpProbe
+}
+
+// BreakHTTPProbe breaks the readiness probe for Nginx StatefulSet containers in ss.
+func BreakHTTPProbe(c clientset.Interface, ss *appsv1.StatefulSet) error {
+	path := httpProbe.HTTPGet.Path
+	if path == "" {
+		return fmt.Errorf("path expected to be not empty: %v", path)
```

Comment:
nit: This `path` is always empty as line 143 and I am not sure it is worth to output it on the error message.
I am not strongly against this at all because this just comes from the original code. Just comment.